### PR TITLE
Praksa

### PR DESCRIPTION
**Pull request by nikolalsvk**

### CODE REVIEW

File: app/views/char_types/_form.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<%= form_for :char_type, url: users_char_types_path, :html => {class: 'form-horizontal', :multipart => true} do |f| %>
-	<% if @char_type.errors.any? %>
+<%= form_for :char_type, url: char_types_path, :html => {class: 'form-horizontal', :multipart => true} do |f| %>
```
Comment (pyromaniackeca) [inline app/views/char_types/_form.html.erb]: class: 'form-horizontal' -> :class => "form-horizontal"

File: app/views/char_types/_form.html.erb [hunk 2]
```diff
@@ -16,14 +16,14 @@
   <div class="field form-group">
     <%= f.label :title, class: 'col-sm-2 control-label' %>
     <div class="col-sm-10">
-    	<%= f.text_field :title, class: 'form-control' %>
- 		</div>
- 	</div> 
- 	
+      <%= f.text_field :title, class: 'form-control' %>
```
Comment (pyromaniackeca) [inline app/views/char_types/_form.html.erb]: class: 'form-control' -> :class => "form-control"

File: config/routes.rb [hunk 3]
```diff
@@ -7,21 +7,16 @@
```
Comment (pyromaniackeca) [inline config/routes.rb:7]: Fix indentation on whole file.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -2,17 +2,17 @@
 require "rails_helper"
```
Comment (pyromaniackeca) [inline spec/models/user_spec.rb:2]: Fix indentation on whole file.

**Comment by nikolalsvk** [comment]
@pyromaniackeca @rastasheep

File: app/controllers/char_types_controller.rb [hunk 5]
```diff
@@ -1,56 +1,53 @@
...
-		@char_type.destroy
-		
-		redirect_to users_char_types_path
-	end
-	
-	private
-		def char_type_params
-			params.require(:char_type).permit(:title, :avatar)
-		end
+  before_action :find_char_type, only: [ :show, :edit, :update, :destroy ]
```
Comment (pyromaniackeca) [inline app/controllers/char_types_controller.rb]: Please use the hash rocket syntax here: :only => [ . . .]

File: app/models/char_attribute.rb [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
+  require 'file_size_validator' 
 
-	mount_uploader :icon, PictureUploader
+  mount_uploader :icon, PictureUploader
   belongs_to :char_type
   
   validates :title, presence: true,
-  									length: { minimum: 3 }
-  									
+                    length: { minimum: 3 }
```
Comment (pyromaniackeca) [inline app/models/char_attribute.rb]: Switch to hash rocket syntax.

File: app/models/char_type.rb [hunk 7]
```diff
@@ -1,21 +1,17 @@
...
-    end	
-  end
+  require 'file_size_validator' 
+  
+  mount_uploader :avatar, PictureUploader
+  belongs_to :user
+  has_many :char_attributes, dependent: :destroy
+  
+  validates :title, presence: true,
+    length: { minimum: 3 }
```
Comment (pyromaniackeca) [inline app/models/char_type.rb]: Switch to hash rocket syntax.

File: app/views/char_attributes/_char_attribute.html.erb [hunk 8]
```diff
@@ -1,16 +1,23 @@
...
+  <td><%= image_tag(char_attribute.icon, :width => 100, :height => 100) if char_attribute.icon.present? %></td>
+  <td><%= link_to 'Show', char_type_char_attribute_path(:char_type_id => char_attribute.char_type.id, 
+                                                        :id => char_attribute.id) %>
+  </td>
+  <td><%= link_to 'Edit', edit_char_type_char_attribute_path(:char_type_id => char_attribute.char_type.id, 
+                                                             :id => char_attribute.id) %>
+  </td>
+  <td><%= link_to 'Destroy', char_type_char_attribute_path(:char_type_id => char_attribute.char_type.id, 
+                                                           :id => char_attribute.id), :method => :delete, 
+                                                           :data => { confirm: 'Are you sure?' },
```
Comment (pyromaniackeca) [inline app/views/char_attributes/_char_attribute.html.erb]: Switch to hash rocket syntax.

File: app/views/char_attributes/_form.html.erb [hunk 9]
```diff
@@ -1,6 +1,7 @@
-<%= form_for [@char_type, @char_type.char_attributes.build], html: { class: "form-horizontal", style: "display:inline;" } do |f| %>
+<%= form_for [@char_type, @char_type.char_attributes.build], html: { :class => "form-horizontal",
+  style: "display:inline;" } do |f| %>
```
Comment (pyromaniackeca) [inline app/views/char_attributes/_form.html.erb]: Switch to hash rocket syntax.

File: features/step_definitions/char_type_manipulation_steps.rb [hunk 10]
```diff
@@ -50,8 +31,4 @@
 Then(/^I should see the "(.*?)" edit page$/) do |arg1|
   page.should have_content(title)
   page.should have_content("Edit")
-end
-
-
-
-
+end
```
Comment (pyromaniackeca) [inline features/step_definitions/char_type_manipulation_steps.rb]: Remove the newline from the end of the file.

File: features/step_definitions/user_restrictions_steps.rb [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+  click_link "All characters"
+end
+
+When(/^I try to view specific character type$/) do
+  visit char_type_path(:id => 1)
+end
+
+When(/^I try to view specific character attribute$/) do
+  visit char_type_char_attribute_path(:char_type_id => 1, :id => 1)
+end
```
Comment (pyromaniackeca) [inline features/step_definitions/user_restrictions_steps.rb]: Remove the newline form the end of the file.

**Comment by pyromaniackeca** [comment]
As far as I'm concerned it's ok for merging when you fix these little details.

File: app/models/char_attribute.rb [hunk 12]
```diff
@@ -1,17 +1,17 @@
 class CharAttribute < ActiveRecord::Base
-	require 'file_size_validator' 
+  require 'file_size_validator' 
 
-	mount_uploader :icon, PictureUploader
+  mount_uploader :icon, PictureUploader
   belongs_to :char_type
   
   validates :title, presence: true,
```
Comment (nikolalsvk) [inline app/models/char_attribute.rb]: This needs to be fixed to => true.

**Pull request merged**